File: model/src/main/java/io/github/chriszhong/model/organization/event/NeedsEventFactory.java
package io.github.chriszhong.model.organization.event;

import io.github.chriszhong.model.organization.relation.Needs;

/**
 * The {@linkplain NeedsEventFactory} interface defines the API for constructing {@linkplain NeedsEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface NeedsEventFactory {

	/**
	 * Constructs an {@linkplain NeedsEvent}.
	 *
	 * @param category
	 *            the {@linkplain EventCategory}.
	 * @param needs
	 *            the {@linkplain Needs}.
	 *
	 * @return a {@linkplain NeedsEvent}.
	 */
	NeedsEvent build(EventCategory category, Needs needs);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/relation/NeedsEventFactory.kt
package runtimemodels.chazm.model.relation

import runtimemodels.chazm.api.relation.Needs
import runtimemodels.chazm.model.event.EventType

/**
 * The [NeedsEventFactory] interface defines the API for constructing [NeedsEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface NeedsEventFactory {
    /**
     * Constructs an [NeedsEvent].
     *
     * @param category the [EventType].
     * @param needs    the [Needs].
     * @return a [NeedsEvent].
     */
    fun build(category: EventType, needs: Needs): NeedsEvent
}
